algorithms/SeatingChart/main.py

```diff
@@ -217,15 +219,67 @@ def generate_seating_charts(room_map_csv, room_allotment_csv, registered_student
                         start_point = start_point ^ 1
 
             for i in range(seated, student_count):
-                print(
-                    f"Could not assign a seat for {course.get_next_student()} in {room} for {course.code} - {course.ic_email}. Assign manually in the room.")
-
+                key_dict = room + "-" + course.time 
+                if key_dict not in left_out_students:
+                    left_out_students[key_dict] = {}
+                if course.code not in left_out_students[key_dict]:
+                    left_out_students[key_dict][course.code] = []
+                
+                left_out_students[key_dict][course.code].append(course.get_next_student())
+                not_alloted_students += 1
+                # print(
+                #     f"Could not assign a seat for {course.get_next_student()} in {room} {course.time} for {course.code} - {course.ic_email}. Assign manually in the room.")
+        left_out_students_copy  = copy.deepcopy(left_out_students)
         if course.allotment_index < len(course.students):
             print(
                 f"Seating Arrangement Discrepancy - {len(course.students) - course.allotment_index} students after {course.get_next_student()} for {course.code} - {course.title}")
 
+    left_out_students_count = 0
+    for time, courses in left_out_students.items():
+
+        room = time.split("-")[0]
+        for course_code in courses:
+
+            count = 0
+            course = course_list.find_by_code(course_code)
+            keys = get_matched_rooms(room_map, room)
+            for key in keys:
+
+                chart = final_solution[course.time][key]
+                limits = room_map[key]
+                for i in range(0, len(limits)):
+
+                    for j in range(start_point, limits[i], step_value):
+
+                        if(count == len(left_out_students[time][course_code])):
+                            break
+
+                        student = left_out_students[time][course_code][count]
+
+                        if chart[len(chart) - j -
+                                1][i] == "":
+                                chart[len(chart) - j -
+                                    1][i] = f"{course.code} - {student}"
+                                left_out_students_copy[time][course_code].remove(student)
+                                count += 1
+                                left_out_students_count += 1
+
+            
     export_charts(room_map, course_list, final_solution)
+    print("Number of not alloted and alloted after taking into account the left_out_students", not_alloted_students, left_out_students_count)
```

Comment:
The meaning of the output message here is not clear. Think of something more concise and clear.